compiler/quilt/nodes.py

```diff
@@ -47,20 +47,30 @@ def __init__(self, package, node, data, meta):
         self._node = node
         self.__cached_data = data
 
-    def _data(self):
+    def _data(self, asa=None):
         """
-        Returns the contents of the node: a dataframe or a file path.
+        Returns the contents of the node: a dataframe or a file path, or passes
+        the node and is contents to a callable.        
         """
-        if self.__cached_data is None:
-            # TODO(dima): Temporary code.
+        if callable(asa):
             store = self._package.get_store()
-            if isinstance(self._node, core.TableNode):
-                self.__cached_data = store.load_dataframe(self._node.hashes)
-            elif isinstance(self._node, core.FileNode):
-                self.__cached_data = store.get_file(self._node.hashes)
-            else:
-                assert False
-        return self.__cached_data
+            return asa(self, [store.object_path(obj) for obj in self._node.hashes])
+        elif asa is not None:
+            raise AttributeError("{asa} is not callable".format(asa=asa))
+        else:
+            if self.__cached_data is None:
```

Comment:
I wonder if the whole `else` clause should be a function used as a default value for `asa`... Though can probably wait till later.